### PR TITLE
Clarify docs for connecting to databricks over odbc

### PR DESCRIPTION
**Pull request by drewbanin**
## Description & motivation
Organization IDs are required for all clouds

## Pre-release docs
Is this change related to an unreleased version of dbt?
- [ ] Yes: please
    - update the base branch to `next`
    - add Changelog components: `<Changelog>[New/Changed] in v0.x.0</Changelog>`
    - add links to the "New and changed documentation" section of the latest [Migration Guide](../website/docs/docs/guides/migration-guide)
- [x] No: please ensure the base branch is `current`
- [ ] Unsure: we'll let you know!

### CODE REVIEW

**Comment by drewbanin** [comment]
@jtcohen6 I would like to actually provide _more info_ as opposed to _less info_ here - please lmk if you think there's a different thing that you think would be more appropriate for us to communicate here!

File: website/docs/docs/dbt-cloud/cloud-configuring-dbt-cloud/connecting-your-database.md [hunk 1]
```diff
@@ -160,7 +160,7 @@ The following fields are available when creating a Databricks connection:
 | ----- | ----------- | -------- |
 | Host Name | The hostname of the Databricks account to connect to | `avc-def1234ghi-9999.cloud.databricks.com` |
 | Port | The port to connect to Databricks for this connection | 443 |
-| Organization | Optional, only required for Azure Databricks | 0123456789 |
+| Organization | Optional | 0123456789 |
```
Comment (jtcohen6) [inline website/docs/docs/dbt-cloud/cloud-configuring-dbt-cloud/connecting-your-database.md]:
```suggestion
| Organization | Optional (default: 0) | 0123456789 |
```

Comment (jtcohen6) [reply website/docs/docs/dbt-cloud/cloud-configuring-dbt-cloud/connecting-your-database.md]: Alternatively:
```
Optional (always 0 for AWS Databricks)
```

**Review by jtcohen6 — COMMENTED** [review]
Yeah I see what you mean. Either of those feel any better?

**Comment by drewbanin** [comment]
yeah! Feels good - I noted that the default value is "zero"

**Pull request merged**